src/clj/game/core/runs.clj

```diff
@@ -349,8 +354,16 @@
                           (make-eid state)
                           {:cancel-fn (fn [state]
                                         (should-end-encounter? state side ice))})
-              (when (should-end-encounter? state side ice)
-                (encounter-ends state side eid)))))
+              (if (should-end-encounter? state side ice)
+                (encounter-ends state side eid)
+                ;; step 6.9.3b: if there are no subroutines on the ice,
+                ;; the runner is considered to have broken all the subroutines on this ice.
+                ;; This should fire an event, so it can get picked up with cards like hippo
+                ;; or knifed.
+                (when-let [c-ice (get-current-ice state)]
+                  (when (and (same-card? c-ice ice) (zero? (count (:subroutines c-ice))))
+                    (wait-for (trigger-event-simult state side :subroutines-broken nil (break-subs-event-context state c-ice [] (get-in @state [:runner :basic-action-card]))))))))))
```

Comment:
no need for an empty `wait-for`, just call the `trigger-event-simult` directly with the eid.